deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -171,14 +171,66 @@ func (c CCIPOCRParams) Validate(selector uint64, feedChainSel uint64, state CCIP
+func WithOCRParamOverride(override func(params *CCIPOCRParams)) CCIPOCROpts {
+	return func(params *CCIPOCRParams) {
+		if override != nil {
+			override(params)
+		}
+	}
+}
+
+// WithDefaultCommitOffChainConfig can be used to add token info to the existing commit off-chain config. If no commit off-chain config is set, it will be created with default values.
+func WithDefaultCommitOffChainConfig(feedChainSel uint64, tokenInfo map[ccipocr3.UnknownEncodedAddress]pluginconfig.TokenInfo) CCIPOCROpts {
+	return func(params *CCIPOCRParams) {
+		if params.CommitOffChainConfig == nil {
+			params.CommitOffChainConfig = &pluginconfig.CommitOffchainConfig{
+				RemoteGasPriceBatchWriteFrequency:  *config.MustNewDuration(internal.RemoteGasPriceBatchWriteFrequency),
+				TokenPriceBatchWriteFrequency:      *config.MustNewDuration(internal.TokenPriceBatchWriteFrequency),
+				TokenInfo:                          tokenInfo,
+				PriceFeedChainSelector:             ccipocr3.ChainSelector(feedChainSel),
+				NewMsgScanBatchSize:                merklemulti.MaxNumberTreeLeaves,
+				MaxReportTransmissionCheckAttempts: 5,
+				RMNEnabled:                         os.Getenv("ENABLE_RMN") == "true", // only enabled in manual test
+				RMNSignaturesTimeout:               30 * time.Minute,
+				MaxMerkleTreeSize:                  merklemulti.MaxNumberTreeLeaves,
+				SignObservationPrefix:              "chainlink ccip 1.6 rmn observation",
+			}
+		} else {
+			if params.CommitOffChainConfig.TokenInfo == nil {
+				params.CommitOffChainConfig.TokenInfo = make(map[ccipocr3.UnknownEncodedAddress]pluginconfig.TokenInfo)
+			}
+			for k, v := range tokenInfo {
+				params.CommitOffChainConfig.TokenInfo[k] = v
+			}
+		}
+	}
+}
+
+// WithDefaultExecuteOffChainConfig can be used to add token data observers to the execute off-chain config. If no execute off-chain config is set, it will be created with default values.
+func WithDefaultExecuteOffChainConfig(tokenDataObservers []pluginconfig.TokenDataObserverConfig) CCIPOCROpts {
+	return func(params *CCIPOCRParams) {
+		if params.ExecuteOffChainConfig == nil {
+			params.ExecuteOffChainConfig = &pluginconfig.ExecuteOffchainConfig{
+				BatchGasLimit:             internal.BatchGasLimit,
+				RelativeBoostPerWaitHour:  internal.RelativeBoostPerWaitHour,
+				InflightCacheExpiry:       *config.MustNewDuration(internal.InflightCacheExpiry),
+				RootSnoozeTime:            *config.MustNewDuration(internal.RootSnoozeTime),
+				MessageVisibilityInterval: *config.MustNewDuration(internal.FirstBlockAge),
+				BatchingStrategyID:        internal.BatchingStrategyID,
+				TokenDataObservers:        tokenDataObservers,
+			}
+		} else if tokenDataObservers != nil {
+			params.ExecuteOffChainConfig.TokenDataObservers = append(params.ExecuteOffChainConfig.TokenDataObservers, tokenDataObservers...)
+		}
+	}
+}
+
+// DeriveCCIPOCRParams derives the default OCR parameters for a chain, with the option to override them.
+func DeriveCCIPOCRParams(
```

Comment:
@makramkd This is a change not related to rename. It will help in overriding the ocr params over default ones. Sample use - https://github.com/smartcontractkit/chainlink-deployments/blob/fee0e6a41d551af62946611d18bff5d0a5958972/ccip/staging/config.go#L145-L161